Geometry/Geometry.h

```diff
@@ -25,173 +27,210 @@ class TGeoManager;
-
-				int NSSDs() const {return (int)fSSD.size(); };
-				Detector GetSSD(int i) const {return fSSD[i]; }
-				double Dz() const { return fDz;}
-				double Width() const { return fWidth; }
-				double Height() const {return fHeight; }
-
-			private:
-				std::string fName;
-				TVector3 fPos;
-				double fDz;
-				double fWidth;
-				double fHeight;
-				std::vector<Detector> fSSD;
-		};
-
-
-		class Geometry {
-			public:
-				Geometry(std::string fname);
-				~Geometry() {};
-
-				bool SetGDMLFile(std::string fname);
-
-				double WorldWidth() const  { return fWorldWidth; }
-				double WorldHeight() const { return fWorldHeight; }
-				double WorldLength() const { return fWorldLength;}
-
-				double MagnetUSZPos() const {return fMagnetUSZPos; }
-				double MagnetDSZPos() const {return fMagnetDSZPos; }
-				bool MagnetLoad() const {return fMagnetLoad; }
-
-				double DetectorUSZPos(int i) const {return fDetectorUSZPos[i]; }
-				double DetectorDSZPos(int i) const {return fDetectorDSZPos[i]; }
-				bool DetectorLoad(int i) const {return fDetectorLoad[i]; }
-
-				int NSSDStations() const { return fNSSDStations; }
-				int NSSDs() const { return fNSSDs; }
-				SSDStation GetSSDStation(int i) {return fSSDStation[i]; }
-
-				int NPMTs() const { return fNPMTs; }
-				emph::arich_util::PMT GetPMT(int i){return fPMT[i]; }
-				emph::arich_util::PMT FindPMTByName(std::string name);
-
-				//    TGeoMaterial* Material(double x, double y, double z) const;
-
-				std::string GDMLFile() const {return fGDMLFile; }
-
-				TGeoManager* ROOTGeoManager() const { return fGeoManager; }
-
-			private:
-				Geometry();
-
-				bool LoadGDMLFile();
-				std::vector<std::pair<double, double> > ReadMatrix(TGDMLMatrix* matrix);
-				void ExtractPMTInfo(const TGeoVolume* v);
-				void ExtractDetectorInfo(int i, const TGeoNode* n);
-				void ExtractMagnetInfo(const TGeoVolume* v);
-				void ExtractSSDInfo(const TGeoNode* n);
-
-				std::string fGDMLFile;
-
-				int    fNSSDStations;
-				int    fNSSDs;
-				double fWorldWidth;
-				double fWorldHeight;
-				double fWorldLength;
-				double fMagnetUSZPos;
-				double fMagnetDSZPos;
-				bool fMagnetLoad;
-				std::vector<SSDStation> fSSDStation;
-				double fDetectorUSZPos[NDetectors];
-				double fDetectorDSZPos[NDetectors];
-				bool fDetectorLoad[NDetectors];
-
-				int    fNPMTs;
-				std::vector<emph::arich_util::PMT> fPMT;
-
-				TGeoManager* fGeoManager;
-
-		};
-
-	}  // end namespace geo
+  namespace geo {
+    enum sensorView {
+      INIT=0,
+      X_VIEW=1, ///< x-measuring view
+      Y_VIEW,   ///< y-measuring view
+      U_VIEW,
+      W_VIEW
+    };
+    class Strip {
+    public:
+      Strip();
+      Strip(std::string name, TVector3 pos, double dw);
+      ~Strip() {};
+
+      std::string Name() const { return fName;}
+      TVector3 Pos() const { return fPos;}
+      double Dw() const { return fDw;}
+      void LocalToMaster(double x1[3], double x2[3]) const { fGeoMatrix->LocalToMaster(x1,x2); }
```

Comment:
On NOvA we made an effort a couple years ago to move away from using 'Master', I'd suggest 'Global' here.

Reply:
Good point.  But Global is misleading, so I've changed it to "LocalToMother".  We still need to call Root's method (LocalToMaster) though.